src/libawkward/array/NumpyArray.cpp

```diff
@@ -2891,10 +2922,11 @@ namespace awkward {
     else {
       std::shared_ptr<Content> out;
       std::shared_ptr<void> ptr;
+      int64_t ptr_length = length();
 
       switch (dtype_) {
       case util::dtype::boolean:
-        ptr = array_sort<bool>(reinterpret_cast<bool*>(data()),
+        std::tie(ptr, ptr_length) = array_sort<bool>(reinterpret_cast<bool*>(data()),
```

Comment:
As far as I've been able to find, this `std::tie` unpacking (to the left of an assignment) is allowed in C++11, so it's fine.

There are a few places where I've used `std::pair` to return two values from a function, such as `minmax_depth`, `branch_depth`, `offsets_and_flatten`, and `nextcarry_outindex`. Those would be easier to read as a `std::tie`, but I didn't know C++ had this feature! Maybe some of these can adopt the `std::tie` technique: `nextcarry_outindex` is only ever used internally, so that's a good candidate. The others are mapped as Python return values by pybind11, so it depends on whether pybind11 recognizes the `std::tie` in the same way that it does `std::pair`. (It probably does, but I don't know for certain.)

Reply:
I was thinking that `std::tie` may be needed to return more than two parameters for a numpy-like `unique` operation. If it's just two, perhaps, it would make sense to return a `ContentPtr` here.